modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java

```diff
@@ -1481,13 +1495,30 @@ public void discardMediaPackage(MediaPackage mp) throws IOException {
     logger.info("Sucessful discarded mediapackage {}", mp);
   }
 
+  /**
+   * Creates a StandAloneTrustedHttpClientImpl
+   *
+   * @param user
+   * @param password
+   * @return
+   */
+  protected TrustedHttpClient createStandaloneHttpClient(String user, String password) {
+    return new StandAloneTrustedHttpClientImpl(this.downloadUser, this.downloadPassword, none(), none(), none());
+  }
+
   protected URI addContentToRepo(MediaPackage mp, String elementId, URI uri) throws IOException {
     InputStream in = null;
     HttpResponse response = null;
+    TrustedHttpClient httpClientStandAlone = httpClient;
     try {
       if (uri.toString().startsWith("http")) {
         HttpGet get = new HttpGet(uri);
-        response = httpClient.execute(get);
+
+        if (uri.getHost().contains(this.downloadSource)) {
```

Comment:
This check means that if credentials are configured for `develop.opencast.org` and an attacker uses a link like `develop.opencast.org.lkiesow.io`, Opencast will send the credentials to that server. Instead `equals()` or a regular expression (if you want to configure multiple servers) should be used.

Reply:
@lkiesow regexp or equals would not do more security. It's still left to the user, for more security the user should wirte source=https://develop.opencast.org/ ....
But i like the idea of adding multiple source with regexp, so done with regexp -> matches